src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -591,24 +567,39 @@ private void applyInjectionContingencies(Network network, LfNetwork lfNetwork, P
         Collection<LfVscConverterStationImpl> vscs = new HashSet<>();
         for (String hvdcId : contingency.getHvdcIdsToOpen()) {
             HvdcLine hvdcLine = network.getHvdcLine(hvdcId);
-            processConverterStation(lfNetwork, hvdcLine.getConverterStation1(), lccs, vscs);
-            processConverterStation(lfNetwork, hvdcLine.getConverterStation2(), lccs, vscs);
+            if (hvdcLine.getConverterStation1() instanceof VscConverterStation) {
+                LfVscConverterStationImpl vsc = (LfVscConverterStationImpl) lfNetwork.getGeneratorById(hvdcLine.getConverterStation1().getId());
+                if (vsc != null) {
+                    vscs.add(vsc);
+                }
+            } else {
+                LfBus bus = lfNetwork.getBusById(hvdcLine.getConverterStation1().getTerminal().getBusView().getBus().getId());
+                if (bus != null) {
+                    lccs.add(Pair.of(bus, (LccConverterStation) hvdcLine.getConverterStation1()));
+                }
+            }
+            if (hvdcLine.getConverterStation2() instanceof VscConverterStation) {
+                LfVscConverterStationImpl vsc = (LfVscConverterStationImpl) lfNetwork.getGeneratorById(hvdcLine.getConverterStation2().getId());
+                if (vsc != null) {
+                    vscs.add((LfVscConverterStationImpl) lfNetwork.getGeneratorById(hvdcLine.getConverterStation2().getId()));
+                }
+            } else {
+                LfBus bus = lfNetwork.getBusById(hvdcLine.getConverterStation2().getTerminal().getBusView().getBus().getId());
+                if (bus != null) {
+                    lccs.add(Pair.of(bus, (LccConverterStation) hvdcLine.getConverterStation2()));
+                }
+            }
         }
 
         // generators.
         Collection<LfGeneratorImpl> generators = new HashSet<>();
-        for (String generatorId : contingency.getGeneratorIdsToLose()) {
-            Generator generator = network.getGenerator(generatorId);
-            processGenerator(lfNetwork, generator, generators);
+        for (Map.Entry<String, Double> e : contingency.getGeneratorIdsToLose().entrySet()) {
+            generators.add((LfGeneratorImpl) lfNetwork.getGeneratorById(e.getKey()));
         }
 
-        for (String loadId : contingency.getLoadIdsToLose()) {
-            Load load = network.getLoad(loadId);
-            Bus bus = load.getTerminal().getBusView().getBus();
-            if (bus != null) {
```

Comment:
What was the reason for null checking here? If load is part of the calculation, it is connected to a bus?